src/tcgwars/logic/groovy/TcgStatics.groovy

```diff
@@ -1146,6 +1146,81 @@ class TcgStatics {
   static boolean isGxPerformed(){
     bg.em().retrieveObject("gx_"+my.owner)
   }
+
+  /* General checks for attacks and abilities */
+
+  /**
+   *
+   * Does a customized assert with an automated fail warning.
+   *
+   * @param params Optional settings that can be added:
+   *   + hasType: If set, restricts to benched Pokémon of a single specific type.
+   *   + hasPokemonEX/hasPokemonGX/hasPokemonV: Can be expanded if needed. All of these unset will have the method search for any Pokémon no matter what, but if even a single one is set true it'll only filter those that are set true as well.
+   *   + opp: If true, checks for the opponent's bench instead of "my" bench.
+   *   + info: If set, it'll replace the end of the failed assert warning with a custom text, instead of the default "follow the stated condition(s)".
+   *   + repText: If true, params.info will override the entirety of the failed assert warning.
+   *
+   * @param c Additional condition the filtered benched Pokémon must follow. Defaults to true (so any benched Pokémon).
+   *
+   */
+  static void assertBench(params=[:], Closure c = null) {
+    def failMessage
+    def checkedBench = params.opp ? opp.bench : my.bench
+
+    def hasPokeCnt = 0
+    [params.hasPokemonEX, params.hasPokemonGX, params.hasPokemonV].each{hasPokeCnt += 1}
+
+    def benchFilter = (
+      if (c == null) { true } else {
+        (
+          if (params.hasType) {it.types.contains(params.hasType)} else {true}
+        ) && (
+          if (!hasPokeCnt) {true} else {
+            (params.hasPokemonEX && it.pokemonEX) ||
+            (params.hasPokemonGX && it.pokemonGX) ||
+            (params.hasPokemonV && it.pokemonV)
+          }
+        ) && (
+          c(it)
+        )
+      }
+    )
+
+    if (params.info + !params.repText) {
+      failMessage = params.info
+    } else {
+      def typeString = param.hasType ? " ${params.hasType.getShortNotation()}" : ""
+
+      def pokeString = "${typeString}Pokémon"
+      if (hasPokeCnt) (
+        pokeString = "${typeString}"
+        i = 1
+        [
+          (params.hasPokemonEX, "Pokémon-EX"),
+          (params.hasPokemonGX, "Pokémon-GX"),
+          (params.hasPokemonV, "Pokémon V")
+        ].each{
+          if (it[0]) {
+            pokeString += it[1] + (if (i == isPokeCnt) "" else if (i == isPokeCnt-1) " or " else ", ")
+            i += 1
+          }
+        }
+      )
+
+      failMessage = "${params.opp ? "Your opponent doesn't" : "You don't"} have any Benched $pokeString that ${params.info ? params.info : "follow the stated condition(s)"}"
+    }
+
+    assert checkedBench.any{benchFilter} : failMessage
```

Comment:
```
benchFilter = {
...
}

assert checkedBench.any(benchFilter) : failMessage
```

Reply:
Just to confirm: You're suggesting benchFilter does not require a def? not sure I'm fully following here.

Reply:
No, it should be def again. what i meant was the curly vs square bracket change.

Reply:
ahh, ok haha